### DISCUSSION TITLE
Some nice(?) ideas for future updates (BIG POST)

### DISCUSSION THREAD

**Discussion by okzerus** [YimMenu/YimMenu#704] [Ideas]
While I was playing with this menu, using almost all of its functions, new ideas constantly came to my mind that would perfectly complement this menu, I think.
I came up with quite a lot of them, so this will be a big post I guess :D
These ideas are JUST ideas, I'm not forcing the dev to add any of them, even if it looks like I do. This menu is free anyways, and there's already too much good stuff in it, so do as you wish, dear Dev.

• **"Phasing" ability for the Vehicle tab.** just like no collision for the player, but for the vehicle. adding a bind for it would be also cool. this came up to me because I was using noclip a lot to fly through the car in front of me while driving around.
• **Assign a head bounty to all players in the session** for the Network > Session. I mean, we already have the "kill everyone in the session" thing, so why not :)
• **Favorite system for the Spawn Vehicle tab and the Player Database.** Featured vehicles will be at the top always unless the player selected a different vehicle category (< not really required). Same thing with the Player Database, but the Remove All button won't affect the favorited players. Adding a star symbol or smth like that would be also nice, but that's just visual and not really necessary.
• **Walk On Water ability in the Self tab.** Just like for vehicles, but for players. Like, who doesn't like to walk on water like Jesus? Not that necessary, you can add this just if you like it and it's easy to make.
• **Modify the vehicle abilities**. Check the boxes to make any vehicle able to open a parachute while in the air, have a nitro boost, or jump.
• **More icon variations for the player list.** "G" for the players with god mode, house icon if they're inside buildings, etc.
• **Teleport as a different tab?** All the teleport "presets" from the Troll tab on the players will be in this tab, they will affect only you. Also, it would be nice to be able to create custom teleport locations at the player's place with different names.
• **"Spawn For" box in the Spawn Vehicle tab.** Checking this box and selecting a player from the list will spawn the vehicle right at the selected player. Just for apologizing, I guess :) (or for REAL fun, if there will be additional box "Spawn For Everyone". Imagine entire session on Khanjali's. WORLD OF TANKS time)
• **Show online players in the Player Database** instead of trying to connect to all of them every time you want to give another lesson to a griefer, check your "friend's" online or whatever.
• **Upgrade context menu!** This thing is UNDERRATED so badly, it can replace almost every type of custom weapon from the Weapons tab and even more.
First, it would be nice if the numbers (1-9) for "switch" and the spacebar for "select" were able to interact with the context menu, making navigating in it much easier.
More features in the context menu - 
Steal Vehicle, Remote Control Vehicle, Repair, Bring for the vehicles;
Remove Weapons, Cage, Freeze(?), Kill for the players;
Force Bodyguard(?) for the peds, so they will attack enemies instead of you. Basically the Wololo! spell :D
Btw, the context menu still freezes after clicking a lot. That's really our problem for rage-click spamming, but there should be some reset button, cause even unchecking the context menu and then checking again doesn't work.
• **Upgrade the Spawn Ped tab.** IT IS actually funny to have an army of bodyguards running around and shooting things you shooting at. Watching big NPC battles with cops or patrolling the city with the FIB agents, attacking the annoying players or griefers is fun!
First, the bodyguards are actually half-bodyguards, cause they're not shooting into other players in any way, even if the player started shooting first. It looks more like a bug, so, yeah.
New weapon type (category) named "PLAYER" and make it set by default. It should have every weapon that the player can have. Cause giving them every single thing that can shoot is still crashing the game. (I don't even know why is this a thing)
More models. Well, maybe it's just me, but my model list is really small. I don't have a lot of singleplayer characters including the main ones, even the Jesus ;c
Reset Model button for the player.

That's everything, I think.
So, just in case, sorry for my stupid humor, too much useless information, and some useless ideas. That's just my fantasy and imagination.

**Comment by 0x00000499** [comment]
您的来信我已收到，将尽快查阅，谢谢！

**Comment by Yimura** [comment]
These are some cool ideas @okzerus, we'll definitely look into adding some of these in the future!

Though be warned, it might take a bit of time before you'll see these become the thing.

On a side-note I definitely agree with your point on the context menu, it should really receive some upgrades. 
Sadly the current version of the context menu appears to be completely broken, it was also made a really long time ago when we had less experience with modding.
